internal/runbits/requirements/requirements.go

```diff
@@ -351,31 +268,198 @@ func (r *RequirementOperation) ExecuteRequirementOperation(
+	requirement.langName = "undetermined"
+
+	if requirement.NamespaceType != nil {
+		switch *requirement.NamespaceType {
+		case model.NamespacePackage, model.NamespaceBundle:
+			commitID, err := localcommit.Get(r.Project.Dir())
+			if err != nil {
+				return errs.Wrap(err, "Unable to get local commit")
+			}
+
+			language, err := model.LanguageByCommit(commitID, r.Auth)
+			if err == nil {
+				requirement.langName = language.Name
+				requirement.Namespace = ptr.To(model.NewNamespacePkgOrBundle(requirement.langName, *requirement.NamespaceType))
+			} else {
+				logging.Debug("Could not get language from project: %v", err)
+			}
+		case model.NamespaceLanguage:
+			requirement.Namespace = ptr.To(model.NewNamespaceLanguage())
+		case model.NamespacePlatform:
+			requirement.Namespace = ptr.To(model.NewNamespacePlatform())
+		}
 	}
 
-	out.Notice(locale.T("operation_success_local"))
+	requirement.validatePkg = requirement.Operation == bpModel.OperationAdded && requirement.Namespace != nil && (requirement.Namespace.Type() == model.NamespacePackage || requirement.Namespace.Type() == model.NamespaceBundle || requirement.Namespace.Type() == model.NamespaceLanguage)
+	if (requirement.Namespace == nil || !requirement.Namespace.IsValid()) && requirement.NamespaceType != nil && (*requirement.NamespaceType == model.NamespacePackage || *requirement.NamespaceType == model.NamespaceBundle) {
+		pg := output.StartSpinner(r.Output, locale.Tr("progress_pkg_nolang", requirement.Name), constants.TerminalAnimationInterval)
+
+		supported, err := model.FetchSupportedLanguages(model.HostPlatform)
+		if err != nil {
+			return errs.Wrap(err, "Failed to retrieve the list of supported languages")
+		}
+
+		var nsv model.Namespace
+		var supportedLang *medmodel.SupportedLanguage
+		requirement.Name, nsv, supportedLang, err = resolvePkgAndNamespace(r.Prompt, requirement.Name, *requirement.NamespaceType, supported, ts, r.Auth)
+		if err != nil {
+			return errs.Wrap(err, "Could not resolve pkg and namespace")
+		}
+		requirement.Namespace = &nsv
+		requirement.langVersion = supportedLang.DefaultVersion
+		requirement.langName = supportedLang.Name
+
+		requirement.validatePkg = false
+
+		pg.Stop(locale.T("progress_found"))
+	}
+
+	if requirement.Namespace == nil {
+		return locale.NewError("err_package_invalid_namespace_detected", "No valid namespace could be detected")
+	}
+
+	return nil
+}
+
+func (r *RequirementOperation) validatePackages(requirements ...*Requirement) error {
+	var requirementsToValidate []*Requirement
+	for _, requirement := range requirements {
+		if !requirement.validatePkg {
+			continue
+		}
+		requirementsToValidate = append(requirementsToValidate, requirement)
+	}
+
+	if len(requirementsToValidate) == 0 {
+		return nil
+	}
+
+	pg := output.StartSpinner(r.Output, locale.Tr("progress_search", strings.Join(requirementNames(requirementsToValidate...), ", ")), constants.TerminalAnimationInterval)
```

Comment:
I wonder if it would be better to have individual progress bars for each requirement, so if there's a failure, it's easy for the user to see which one failed.

Reply:
Initially I had that but it doesn't work with the mock